numpy/core/src/private/lowlevel_strided_loops.h

```diff
@@ -757,15 +752,22 @@ PyArray_EQUIVALENTLY_ITERABLE_OVERLAP_OK(PyArrayObject *arr1, PyArrayObject *arr
     return (!arr1_read || arr1_ahead) && (!arr2_read || arr2_ahead);
 }
 
+#define PyArray_EQUIVALENTLY_ITERABLE_BASE(arr1, arr2) (            \
+                        PyArray_NDIM(arr1) == PyArray_NDIM(arr2) && \
+                        PyArray_CompareLists(PyArray_DIMS(arr1), \
+                                             PyArray_DIMS(arr2), \
+                                             PyArray_NDIM(arr1)) && \
+                        (PyArray_FLAGS(arr1)&(NPY_ARRAY_C_CONTIGUOUS| \
+                                      NPY_ARRAY_F_CONTIGUOUS)) & \
```

Comment:
`&` <- `&&` ?

Reply:
You mean to make it an `&&`? The current version is correct though tricky, since the contiguity has to be identical, frankly would be better to read something like
```
PyArray_FLAGS(arr1) & PyArray_FLAGS(arr2) &
    (NPY_ARRAY_C_CONTIGUOUS|NPY_ARRAY_F_CONTIGUOUS)
```
to be more obvious probably

Reply:
Or maybe not, can't make up my mind :)

Reply:
Right now, I think this means that they need to share at least one contiguity - that sounds plausibly correct to me

Reply:
Ah, OK, make sense.